Assets/Scripts/Core/Templates/MonoBehaviourExtended.cs

```diff
@@ -8,6 +8,9 @@
 
 public abstract class MonoBehaviourExtended : MonoBehaviour, System.IComparable 
 {
+	protected const int NONE_VALUE = 0;
+	protected const int INVALID_VALUE = -1;
+
```

Comment:
These aren't meant to cover all edge cases. They're constants to eliminate needing to write -1 and 0 in subclasses. And it reduces the amount of hardcoded vars in the subclasses. I find that it's more expressive, e.g. in a function like:

```
int findIndex(string[] elements, string target)
{
     if(found)
    {
           return index;
     }
     else
     {
         return INVALID_VALUE;
     }
}